packages/compass-editor/src/ace/validation-autocompleter.ts

```diff
@@ -34,16 +33,29 @@ class ValidationAutoCompleter implements Ace.Completer {
     // we want to suggest document fields instead of suggesting operators.
     const currentToken = session.getTokenAt(position.row, position.column);
     if (currentToken?.type === 'string') {
-      const strings = ([] as CompletionWithServerInfo[]).concat(
-        BSON_TYPE_ALIASES,
-        this.fields
+      return done(
+        null,
+        completer(prefix, {
+          serverVersion: this.version,
+          fields: this.fields
+            .filter(
+              (field): field is CompletionWithServerInfo & { name: string } =>
+                !!field.name
+            )
+            .map((field) => field.name),
```

Comment:
Just a thought, these lines show up a few times here, maybe it makes sense to deduplicate them into a function? Definitely not a big deal though

Reply:
Totally, we already have a `utils` file in the package, might as well put it there 😄

Reply:
Done in [d1e53e1](https://github.com/mongodb-js/compass/pull/4096/commits/d1e53e1811500984b6fcb93dc10c5492b27a1f02)